### PR TITLE
ENH: pd.NamedAgg forwards *args and **kwargs to aggfunc

### LINKED ISSUE #58283: ENH: pd.NamedAgg should accept args and kwargs
### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

This part of the API seems inconsistent:

```python
In [24]: df = pd.DataFrame({"group": list("aab"), "val1": range(3)})
In [28]: def n_between(ser, low, high):
    ...:     return ser.between(low, high).sum()
    ...: 

In [29]: df.groupby("group")["val1"].agg(n_between, 0, 1)  # works
Out[29]: 
group
a    2
b    0
Name: val1, dtype: int64

In [30]: df.groupby("group").agg(n_between=pd.NamedAgg("val1", n_between, 0, 1))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[30], line 1
----> 1 df.groupby("group").agg(n_between=pd.NamedAgg("val1", n_between, 0, 1))

TypeError: NamedAgg.__new__() takes 3 positional arguments but 5 were given
```

### Feature Description

NamedAgg should forward along *args and **kwargs just like the normal callable function application can

### Alternative Solutions

status quo

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by tomytp**
- [x] closes #58283
- [x] [Tests added and passed]
- [x] All [code checks passed]
- [x] Added an entry in the latest `doc/source/whatsnew/v3.0.0.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/user_guide/style.ipynb
Comment (WillAyd) [inline doc/source/user_guide/style.ipynb:1]: Are changes to this file relevant at all?

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -108,7 +108,12 @@
 ScalarResult = TypeVar("ScalarResult")
 
 
-class NamedAgg(NamedTuple):
+class _BaseNamedAgg(NamedTuple):
```
Comment (WillAyd) [inline pandas/core/groupby/generic.py:111]: What's the reason for introducing a new class here?

File: pandas/core/groupby/generic.py [hunk 2]
```diff
@@ -134,8 +143,23 @@ class NamedAgg(NamedTuple):
...
+    @staticmethod
+    def _get_wrapped_aggfunc(function, *initial_args, **initial_kwargs):
+        def wrapped_aggfunc(*new_args, **new_kwargs):
+            final_args = new_args + initial_args
+            final_kwargs = {**initial_kwargs, **new_kwargs}
+            return function(*final_args, **final_kwargs)
+
+        return wrapped_aggfunc
+
+    def __repr__(self):
```
Comment (WillAyd) [inline pandas/core/groupby/generic.py:161]: Is this related to the args/kwargs forwarding?

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (tomytp) [reply pandas/core/groupby/generic.py:111]: That's because you cannot override __new__ when inheriting from NamedTuple directly.

File: pandas/core/groupby/generic.py (hunk 2 above)
Comment (tomytp) [reply pandas/core/groupby/generic.py:161]: The __repr__ is unnecessary, it related to a previous solution that I did not use. I will rollback this change.

File: doc/source/user_guide/style.ipynb
Comment (tomytp) [reply doc/source/user_guide/style.ipynb:1]: I believe this was made automatically by the IDE linter. I'll rollback this file

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (tomytp) [reply pandas/core/groupby/generic.py:111]: So the only way I found to override it while maintaining the NamedTuple properties was to Introduce a new class. Do you have a suggestion for another method to do it?

Comment (tomytp) [reply pandas/core/groupby/generic.py:111]: the reference error for when I tried to inherit directly from NamedTuple is:
`AttributeError: Cannot overwrite NamedTuple attribute __new__`

Comment (WillAyd) [reply pandas/core/groupby/generic.py:111]: Ah that's unfortunate. This seems kind of hacky as a result though - maybe we can just drop the NamedTuple subclass though? I think that was mostly as a convenience rather than an intentional design decision.

@rhshadrach might have more thoughts

Comment (tomytp) [reply pandas/core/groupby/generic.py:111]: Apply does expect a tuple, so to remove this behavior we would need to make abigger change.

Comment (rhshadrach) [reply pandas/core/groupby/generic.py:111]: https://github.com/python/cpython/issues/77258 has some relevant information, especially the linked discussion there to the issue in `python/typing`.

In any case, it looks to me that this implementation just ignores `args` / `kwargs` when the provided `aggfunc` is not a callable? If that's the case, then I'm -1 on this approach anyways.

Comment (tomytp) [reply pandas/core/groupby/generic.py:111]: @rhshadrach I'll rethink the approach to include the function names passed as string.

I'm not sure I understand your thoughts on the necessity of keeping NamedArgs with tuple behavior. Could you clarify that? 

TY

Comment (rhshadrach) [reply pandas/core/groupby/generic.py:111]: > I'm not sure I understand your thoughts on the necessity of keeping NamedArgs with tuple behavior. Could you clarify that?

I don't believe I shared any thoughts in this regard, just a link to some background information on why Python raises when you try to overwrite `__new__`.

**Comment by rhshadrach** [comment]
Closing as stale. @tomytp - if you'd like to continue, comment here and we can reopen.
